### PR TITLE
Add Kotlin names to the node.

### PR DESCRIPTION
**Pull request by SHeimlich**
Currently graph2schema sets the entity class name using the "name" field of the node. We want to update this name to be different for Kotlin schemas, but in the shorter term are less concerned with C++. To transition Kotlin first, we need a way of updating the kotlin names, but leaving the c++ names on the old system. To do this, we add `kotlinName` to the node.

In a similar manner, the alias array tracks what type aliases should be given to the different classes. This also will be updating in Kotlin (before c++) and thus we add `kotlinAliases` to track this as well.

### CODE REVIEW

File: src/tools/schema2graph.ts [hunk 1]
```diff
@@ -71,11 +73,12 @@ export class SchemaGraph {
     }
   }
 
-  private createNodes(schema: Schema, name: string) {
+  private createNodes(schema: Schema, name: string, kotlinName: string) {
     let node = this.nodes.find(n => schema.equals(n.schema));
     if (node) {
-      // We can only have one node in the graph per schema. Collect duplicates as aliases.
+      // We can only have one nosde in the graph per schema. Collect duplicates as aliases.
```
Comment (piotrswigon) [inline src/tools/schema2graph.ts]: nosde?

File: src/tools/schema2graph.ts [hunk 2]
```diff
@@ -19,7 +19,9 @@ export class SchemaNode {
   // is empty. Otherwise, name is of the form 'ParticleInternal#' and aliases lists the
   // 'Particle_Handle' names that need to be type aliased to it.
   name: string;
+  kotlinName: string;
```
Comment (piotrswigon) [inline src/tools/schema2graph.ts]: Can you add a comment for what this is intended to be? The comment above for 'name' is quite useful.

**Comment by piotrswigon** [comment]
I don't understand why you need kotlinName and kotlinAliases, can you provide some description for this patch so that the reviewers have necessary context?

**Comment by SHeimlich** [comment]
> I don't understand why you need kotlinName and kotlinAliases, can you provide some description for this patch so that the reviewers have necessary context?

I updated the PR description. TL;DR is I'd like to update the Kotlin names without updating the C++ names (for now).

File: src/tools/schema2graph.ts (hunk 2 above)
Comment (SHeimlich) [reply src/tools/schema2graph.ts]: Moved the Kotlin variables to be together and added a comment.

**Pull request merged**